tests/backup/backup_test.go

```diff
@@ -549,6 +549,195 @@ var _ = Describe("{BasicBackupCreation}", func() {
 	})
 })
 
+var _ = Describe("{BasicSelectiveRestore}", func() {
+	var (
+		appList           = Inst().AppList
+		backupName        string
+		contexts          []*scheduler.Context
+		preRuleNameList   []string
+		postRuleNameList  []string
+		appContexts       []*scheduler.Context
+		bkpNamespaces     []string
+		clusterUid        string
+		clusterStatus     api.ClusterInfo_StatusInfo_Status
+		restoreName       string
+		cloudCredName     string
+		cloudCredUID      string
+		backupLocationUID string
+		bkpLocationName   string
+		numDeployments    int
+	)
+	backupLocationMap := make(map[string]string)
+	labelSelectors := make(map[string]string)
+	bkpNamespaces = make([]string, 0)
+	selectedBkpNamespaces := make([]string, 0)
+	namespaceMapping := make(map[string]string)
+	providers := getProviders()
+	intervalName := fmt.Sprintf("%s-%v", "interval", time.Now().Unix())
+	dailyName := fmt.Sprintf("%s-%v", "daily", time.Now().Unix())
+	weeklyName := fmt.Sprintf("%s-%v", "weekly", time.Now().Unix())
+	monthlyName := fmt.Sprintf("%s-%v", "monthly", time.Now().Unix())
+
+	if len(appList) == 1 {
+		numDeployments = 6
+	} else {
+		numDeployments = Inst().GlobalScaleFactor
+	}
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("BasicSelectiveRestore", "All namespace backup and restore selective namespaces", nil, 0)
+		log.InfoD("Verifying if the pre/post rules for the required apps are present in the AppParameters or not ")
+		for i := 0; i < len(appList); i++ {
+			if Contains(postRuleApp, appList[i]) {
+				if _, ok := portworx.AppParameters[appList[i]]["post"]; ok {
+					dash.VerifyFatal(ok, true, "Post Rule details mentioned for the apps")
+				}
+			}
+			if Contains(preRuleApp, appList[i]) {
+				if _, ok := portworx.AppParameters[appList[i]]["pre"]; ok {
+					dash.VerifyFatal(ok, true, "Pre Rule details mentioned for the apps")
```

Comment:
same here

Reply:
I realized I am not using any pre or post rules. Removed any related code.